pkg/endpoint/endpoint.go

```diff
@@ -2308,3 +2308,11 @@ func (e *Endpoint) setDefaultPolicyConfig() {
 func (e *Endpoint) GetCreatedAt() time.Time {
 	return e.createdAt
 }
+
+// StopEventQueue stops the event queue of this endpoint.
+func (e *Endpoint) StopEventQueue() {
+	e.eventQueue.Stop()
+	e.getLogger().Debug("eventQueue stopped")
+	e.eventQueue.WaitToBeDrained()
+	e.getLogger().Debug("eventQueue drained")
```

Comment:
We already do this in `Endpoint.Stop()` and we already call that function on agent shutdown: https://github.com/cilium/cilium/blob/141cd637d8fc028e46d6912bb6d852ad89779377/daemon/cmd/daemon.go#L407-L412

Why is that not enough? Maybe it's just a matter of adding a WaitGroup for those stops?

Reply:
Oops, sorry I didn't realize we already do this in master.
Yes, I'll add a WaitGroup to the existing function.